### PR TITLE
Download CRC installer binary on runtime

### PR DESCRIPTION
**Pull request by evidolob**
The move CRC extension to separate repo force us to downloading CRC installation binary in runtime, after user click install button.

> Note: PR uses status bar item to indicate downloading process, that should be replaced when https://github.com/containers/podman-desktop/issues/1254 and https://github.com/containers/podman-desktop/issues/1343 will be implemented

Demo Windows:

https://user-images.githubusercontent.com/929743/224680849-cb05b623-12d0-4fb3-a570-18c450aa5d1f.mp4

Demo MacOS:

https://user-images.githubusercontent.com/929743/224681387-76f9358a-ee35-4a58-9df0-46756b7ade05.mp4

### CODE REVIEW

File: src/install/base-install.ts [hunk 1]
```diff
@@ -34,26 +44,136 @@ export abstract class BaseCheck implements extensionApi.InstallCheck {
...
+
+    this.statusBarItem.text = 'Downloading';
+    const downloadStream =  got.stream(installerUrl);
+
+    
+    downloadStream.on('downloadProgress', (progress) => {
+      const progressStr = /* progress.transferred + ' of ' + progress.total + ' ' +  */Math.round(progress.percent * 100) + '%';
+      if(lastProgressStr !== progressStr) {
+        //TODO: show progress on UI!
+        this.statusBarItem.text = 'Downloading: ' + progressStr;
```
Comment (gbraad) [inline src/install/base-install.ts:116]: This is allowed ;-), but the rest of the API is restrictive...

File: src/install/mac-install.ts [hunk 2]
```diff
@@ -16,32 +16,33 @@
...
+  install(releaseInfo: CrcReleaseInfo, logger?: extensionApi.Logger): Promise<boolean> {
     return extensionApi.window.withProgress({ location: extensionApi.ProgressLocation.APP_ICON }, async progress => {
       progress.report({ increment: 5 });
+      
+      const pkgPath = await this.downloadAndCheckInstaller(releaseInfo.links.darwin, macosInstallerFineName);
 
-      const pkgPath = path.resolve(getAssetsFolder(), 'crc-macos-installer.pkg');
       try {
-        if (fs.existsSync(pkgPath)) {
+        if (await isFileExists(pkgPath)) {
```
Comment (gbraad) [inline src/install/mac-install.ts:37]: is this related to the main ui async issue you mentioned?

Comment (evidolob) [reply src/install/mac-install.ts:37]: yes, I made a lot of `sync` code clean up, ensuring that we don’t use any synchronous code/IO operation.

**Comment by deboer-tim** [comment]
Not expecting it here, but is it possible to switch to silent (background) installer in the future?

**Pull request merged**